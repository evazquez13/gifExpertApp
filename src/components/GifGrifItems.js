import React from 'react';
import PropTypes from 'prop-types';

export const GifGrifItems = ({title,url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGrifItems.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired    
} 