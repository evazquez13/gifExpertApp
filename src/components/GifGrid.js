import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGrifItems } from './GifGrifItems';

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeInDownBig">{category}</h3>

            {loading && <p className="animate__animated animate__flash">Loading</p>}
            
            <div className="card-grid animate__animated animate__flipInY">
                {
                    data.map( img => (<GifGrifItems
                                            key={img.id}
                                            {...img}/>))
                }
                
            </div>
        </>
    )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired
}
