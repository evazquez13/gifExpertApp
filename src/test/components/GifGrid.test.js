import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas al componente <GifGrid/>', () =>{

    const category = 'Goku';
    
    test('Validar que el componente se muestre de manera correcta', () =>{

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar iimagenes cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id:'ABC',
            url:'http://localhost/cualquier/cosa.jpg',
            title:'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGrifItems').length).toBe(gifs.length);
        
    });

});