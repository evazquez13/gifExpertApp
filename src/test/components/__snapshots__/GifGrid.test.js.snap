// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGrid/> Debe mostrar iimagenes cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDownBig"
  >
    Goku
  </h3>
  <div
    className="card-grid animate__animated animate__flipInY"
  >
    <GifGrifItems
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas al componente <GifGrid/> Validar que el componente se muestre de manera correcta 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDownBig"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__flipInY"
  />
</Fragment>
`;
